test/hotspot/jtreg/runtime/cds/CDSMapTest.java

```diff
@@ -46,34 +49,56 @@ public static void main(String[] args) throws Exception {
     }
 
     public static void doTest(boolean compressed) throws Exception {
-        ArrayList<String> dumpArgs = new ArrayList<>();
+        ArrayList<String> vmArgs = new ArrayList<>();
 
         // Use the same heap size as make/Images.gmk
-        dumpArgs.add("-Xmx128M");
+        vmArgs.add("-Xmx128M");
 
         if (Platform.is64bit()) {
             // These options are available only on 64-bit.
             String sign = (compressed) ?  "+" : "-";
-            dumpArgs.add("-XX:" + sign + "UseCompressedOops");
+            vmArgs.add("-XX:" + sign + "UseCompressedOops");
         }
 
-        dump(dumpArgs);
+        String archiveFile = dump(vmArgs);
+        exec(vmArgs, archiveFile);
+
     }
 
     static int id = 0;
-    static void dump(ArrayList<String> args, String... more) throws Exception {
+
+    // Create a map file when creating the archive
+    static String dump(ArrayList<String> args) throws Exception {
         String logName = "SharedArchiveFile" + (id++);
         String archiveName = logName + ".jsa";
         String mapName = logName + ".map";
         CDSOptions opts = (new CDSOptions())
             .addPrefix("-Xlog:cds=debug")
             .addPrefix("-Xlog:aot+map=debug,aot+map+oops=trace:file=" + mapName + ":none:filesize=0")
             .setArchiveName(archiveName)
-            .addSuffix(args)
-            .addSuffix(more);
+            .addSuffix(args);
         CDSTestUtils.createArchiveAndCheck(opts);
 
         CDSMapReader.MapFile mapFile = CDSMapReader.read(mapName);
         CDSMapReader.validate(mapFile);
+
+        return archiveName;
+    }
+
+    // Create a map file when using the archive
+    static void exec(ArrayList<String> vmArgs, String archiveFile) throws Exception {
+        String mapName = archiveFile + ".exec.map";
+        vmArgs.add("-XX:SharedArchiveFile=" + archiveFile);
+        vmArgs.add("-Xlog:cds=debug");
+        vmArgs.add("-Xshare:on");
+        vmArgs.add("-Xlog:aot+map=debug,aot+map+oops=trace:file=" + mapName + ":none:filesize=0");
```

Comment:
Looks like this test is intended for old CDS archive. Do you have test for new AOTCache workflow?

Reply:
I added a new test case for the AOTCache workflow.